lib/rules/no-extra-semi.js

```diff
@@ -34,7 +40,9 @@ module.exports = {
                 node: nodeOrToken,
                 message: "Unnecessary semicolon.",
                 fix(fixer) {
-                    return fixer.remove(nodeOrToken);
+                    return new FixTracker(fixer, context.getSourceCode())
```

Comment:
Could you add a comment here with an explanation and/or a link to the relevant issue? Until we implement a more general solution to solving these conflicts, I don't want someone to look at this code and copy it somewhere else with the assumption that it's standard boilerplate for autofixing.